### PR TITLE
SN1-117 - move offline indexer to the main project

### PR DESCRIPTION
**Pull request by aph5nt**

### CODE REVIEW

File: neurons/nodes/bitcoin/node.py
[the remark this answers is not in the document]
Comment (aph5nt) [reply neurons/nodes/bitcoin/node.py:1]: i see, how much disk are taking those pickles? maybe we could have them splited into multimple files and load pickled on demand? im thinking about some easo of use or miners

[the remark this answers is not in the document]
Comment (aph5nt) [reply neurons/nodes/bitcoin/node.py:1]: another question... will it be still easy for you to use it without docker ?

[the remark this answers is not in the document]
Comment (ronnie-samaroo) [reply neurons/nodes/bitcoin/node.py:1]: image

[the remark this answers is not in the document]
Comment (ronnie-samaroo) [reply neurons/nodes/bitcoin/node.py:1]: > another question... will it be still easy for you to use it without docker ?

yes, it's easy to use without docker.

[the remark this answers is not in the document]
Comment (ronnie-samaroo) [reply neurons/nodes/bitcoin/node.py:1]: I think we can load pickles dynamically at runtime so users just need to specify a single directory.

**Review by ronnie-samaroo — APPROVED** [review]

File: neurons/nodes/bitcoin/btc-vout-hashtable-builder/utils.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    # return pairs of start_pos and end_pos
+    result = []
+    for i in range(len(positions)):
+        start_pos = positions[i]
+        end_pos = positions[i + 1] if i < len(positions) - 1 else -1
+        result.append((start_pos, end_pos))
+    return result
+
+
+def get_sub_keys():
```
Comment (0xMarx) [inline neurons/nodes/bitcoin/btc-vout-hashtable-builder/utils.py:37]: duplicate method node_utils get_tx_out_hash_table_sub_keys()

**Review by 0xMarx — CHANGES_REQUESTED** [review]

**Review by 0xMarx — APPROVED** [review]

**Pull request merged**